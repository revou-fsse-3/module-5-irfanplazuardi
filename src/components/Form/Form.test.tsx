import { render, fireEvent, screen, waitFor } from "@testing-library/react";
import FormCity from "@/components/Form";

describe("Form unit test", () => {
  test("renders form", () => {
    const mockOnSubmit = jest.fn();
    render(<FormCity onSubmit={mockOnSubmit} />);

    const labelElement = screen.getByText("City:");
    expect(labelElement).toMatchSnapshot();

    const inputElement = screen.getByRole("textbox");
    expect(inputElement).toMatchSnapshot();

    const buttonElement = screen.getByText("Submit");
    expect(buttonElement).toMatchSnapshot();
  });

  test("submit form", async () => {
    const mockOnSubmit = jest.fn();
    render(<FormCity onSubmit={mockOnSubmit} />);
    const input = screen.getByRole("textbox");
    const button = screen.getByRole("button", { name: "Submit" });
    fireEvent.change(input, { target: { value: "New York" } });
    fireEvent.click(button);

    await waitFor(() => {
      expect(mockOnSubmit).toHaveBeenCalledWith({ city: "New York" });
    });
  });

  test("submit form with error", async () => {
    const mockOnSubmit = jest.fn();
    render(<FormCity onSubmit={mockOnSubmit} />);

    const button = screen.getByRole("button", { name: "Submit" });
    fireEvent.click(button);

    await waitFor(() => {
      expect(mockOnSubmit).not.toHaveBeenCalled();

      expect(screen.getByText("Please enter a city")).toMatchSnapshot();
    });
  });
});
