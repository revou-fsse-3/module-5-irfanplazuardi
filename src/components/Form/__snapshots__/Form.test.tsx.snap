// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form unit test renders form 1`] = `
<label
  for="city"
>
  City:
</label>
`;

exports[`Form unit test renders form 2`] = `
<input
  id="city"
  name="city"
  type="text"
  value=""
/>
`;

exports[`Form unit test renders form 3`] = `
<button
  type="submit"
>
  Submit
</button>
`;

exports[`Form unit test submit form with error 1`] = `
<div>
  Please enter a city
</div>
`;
